numba/typing/npydecl.py

```diff
@@ -62,13 +62,27 @@ def _handle_inputs(cls, ufunc, args, kws):
             raise TypingError(msg=msg.format(ufunc.__name__))
 
         # find the kernel to use, based only in the input types (as does NumPy)
-        base_types = [x.dtype if isinstance(x, types.Array) else x for x in args]
-        return base_types, explicit_outputs, ndims
+        base_types = [x.dtype if isinstance(x, types.Array) else x
+                      for x in args]
+
+        # Figure out the output array layout, if needed.
+        layout = None
+        if ndims > 0:
+            layout = 'C'
+            layouts = [x.layout if isinstance(x, types.Array) else ''
+                       for x in args]
+            if 'C' not in layouts:
+                if 'F' in layouts:
+                    layout = 'F'
+                elif 'A' in layouts:
+                    layout = 'A'
```

Comment:
Just for the record, output layout 'A' isn't implemented, right?
Perhaps this should simply be a TypingError?